specification/programmableconnectivity/ProgrammableConnectivity.Management/Gateway.tsp

```diff
@@ -51,6 +51,312 @@ model GatewayProperties {
    */
   @visibility("read")
   provisioningState?: ProvisioningState;
+
+  /**
+   * Details about the Application that would use the Operator's Network APIs.
+   */
+  @TypeSpec.Versioning.added(Versions.v2024_11_01_preview)
+  @visibility("read", "create", "update")
+  configuredApplication: ApplicationProperties;
+
+  /**
+   * Details about the Organization owning the Application that would use the Operator's Network APIs.
+   */
+  @TypeSpec.Versioning.added(Versions.v2024_11_01_preview)
+  @visibility("read", "create", "update")
+  configuredApplicationOwner: ApplicationOwnerProperties;
+}
+
+/**
+ * Details about the Application Owner (i.e. organization) that would use the Operator's Network APIs.
+ */
+@TypeSpec.Versioning.added(Versions.v2024_11_01_preview)
+model ApplicationOwnerProperties {
+  /**
+   * Common name of the organization owning the application.
+   */
+  name: string;
+
+  /**
+   * Legal name of the organization owning the application.
+   */
+  legalName: string;
+
+  /**
+   * Trading name of the organization owning the application.
+   */
+  tradingName: string;
+
+  /**
+   * A description of the organization owning the application.
+   */
+  organizationDescription: string;
+
+  /**
+   * Unique Tax Number for the user's organization in the country/region the APC Gateway is being purchased.
+   */
+  taxNumber: string;
+
+  /**
+   * The type of organization that the Application Owner is.
+   */
+  organizationType: OrganizationType;
```

Comment:
I'm wondering if using an Enum here is the right approach. There's a possibility that future TMF931 changes will add values to this Enum, without changing anything else that would have an impact on our ARM API - that would force us to do a whole new ARM API version just to add a few more values to this. On the flip side, if TMF931 changes to remove values, we'd have to maintain them for 3 years returning an error.

We can enforce the set of permitted values more easily and flexibly using validation in our RP. 

I'm not sure if there's a clear user value to making this Enum part of the API spec - we'd be promising a bit more stability to the users that fields won't be removed from the Enum, but that's not really a promise we can make, given that it's controlled by TMF931 anyway.

Not a strong opinion so happy to discuss

Reply:
ARM favors enum over unrestricted string (or boolean) whenever possible. Enums generally make better properties, since values are somewhat self-documenting (and can have descriptions). As you observe, there is a tradeoff for properties whose allowed values change quickly: it's easier for us to validate at the service. However, removing values is not qualitatively different from a client perspective than changing allowed string values in the backend: both types of changes are breaking and will require clients to update their code. A benefit of doing it in the schema is that clients can know a priori which values are allowed for a given API version. This gives them more control over when to update their code (only needed to support new API versions). If it can change at any time even between API versions, this leaves customers in a position of needing to respond immediately whenever we randomly decide to break their clients.

Reply:
I think that having these as ENUMs makes using them much easier, as it's immediately clear to the user what values are acceptable. I also expect it means that Portal will be updated with new values when integrating with a new version, without having to track all the possible values in 2 places. 

I also can't see TMForum removing any fields from these lists. If TMF add fields, we can decide whether those are worth a new ARM API version.